src/java/com/unifina/signalpath/ModuleExceptionMessage.java

```diff
@@ -1,18 +1,19 @@
 package com.unifina.signalpath;
 
-import java.util.List;
 import java.util.Map;
 
-public class ModuleExceptionMessage {
-	public ModuleExceptionMessage(int hash, Map<String, Object> msg) {
+public abstract class ModuleExceptionMessage {
+	protected int hash;
+	protected Map<String,Object> msg;
+
+	ModuleExceptionMessage(int hash, Map<String, Object> msg) {
```

Comment:
The goal is not to pass any `Map`s around anymore. `msg` is not needed here at all. Enough for this superclass to just have a constructor with `ModuleExceptionMessage(int hash)`, then leave it up to the children what else they might contain (and how to render themselves `toMap`).